IQSS/dataverse
# extract metadata (NcML XML) from NetCDF/HDF5 files, new "requirements" option for external tools

**What this PR does / why we need it**:

From talking to researchers who use NetCDF and HDF5 files, we believe there is value in extracting metadata from those files to give some insight about their contents.

I plan to also make a pull request in the https://github.com/gdcc/dataverse-previewers repo to add a preview tool for the extracted file, which is in NcML format.

**Which issue(s) this PR closes**:

- Closes #9153
- Relates to #9242

**Special notes for your reviewer**:

Additional scope we could consider:

- an API endpoint to extract NcML from existing NetCDF and HDF5 files
- have the "requirements" option apply in more places than file level preview tools
- making the extraction feature modular

**Suggestions on how to test this**:

- try various NetCDF and HDF5 files and see if NcML aux files are created
- try the new "requirements" option for external tools
- use the proposed NcML previewer in this PR: https://github.com/gdcc/dataverse-previewers/pull/18

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

No but here's an example of two HDF5 files where one doesn't show the eyeball because it doesn't meet the requirement of having a aux file:

![Screen Shot 2022-12-20 at 10 05 08 AM](https://user-images.githubusercontent.com/21006/208703896-716111f8-909f-44e1-9135-ee924916618b.png)

**Is there a release notes update needed for this change?**:

Yes, included.

**Additional documentation**:

Included.
